test/integration/test_push_docker_image.py

```diff
@@ -0,0 +1,109 @@
+import contextlib
+import docker
+import json
+import logging
+import pytest
+import re
+import requests
+import time
+
+from exasol.ds.sandbox.lib.dss_docker import DssDockerImage
+
+from test.integration.local_docker_registry import (
+    custom_docker_registry_context,
+    LocalDockerRegistry,
+    local_docker_registry_context,
+)
+
+_logger = logging.getLogger(__name__)
+_logger.setLevel(logging.DEBUG)
+
+
+def normalize_request_name(name: str):
+    name = re.sub(r"[\[\]._]+", "_", name)
+    return re.sub(r"^_+|_+$", "", name)
+
+
+@contextlib.contextmanager
+def tagged_image(
+        dss_docker_image: DssDockerImage,
+        registry: LocalDockerRegistry,
+        tag: str = None,
+):
+    """
+    Prepend host and port of LocalDockerRegistry to the repository name of the
+    DssDockerImage to enable pushing the image to the local registry.
+    """
+    new_repo = f"{registry.host_and_port}/{dss_docker_image.repository}"
+    client = docker.from_env()
+    tag = tag or dss_docker_image.version
+    _logger.info(f'tagging image to {new_repo}:{tag}')
+    image = client.images.get(dss_docker_image.image_name)
+    image.tag(new_repo, tag)
+    tagged = DssDockerImage(new_repo, tag)
+    tagged.registry = registry
+    try:
+        yield tagged
+    finally:
+        docker.from_env().images.remove(tagged.image_name)
+
+
+@pytest.fixture(scope="session")
+def docker_registry(request):
+    """
+    Provide a context for creating a LocalDockerRegistry accepting
+    parameter ``repository``.
+
+    You can provide cli option ``--docker-registry HOST:PORT`` to pytest in
+    order reuse an already running Docker container as registry.
+    """
+    existing = request.config.getoption("--docker-registry")
+    if existing is not None:
+        yield LocalDockerRegistry(existing)
+        return
+
+    test_name = normalize_request_name(request.node.name)
+    container_name = f"{test_name}_registry"
+
+    port = find_free_port()
+    client = docker.from_env()
+    _logger.debug("Pulling Docker image with Docker registry")
+    client.images.pull(repository="registry", tag="2")
```

Comment:
You might want to pull only if the image is not already available locally

Reply:
As far as I did observe, the `client.images.pull` is smart enough to only pull the image if not available locally. 
Should I change it, though?